### ISSUE TITLE
Rework SCA Policy for Microsoft Windows Server 2012 R2

### ISSUE THREAD

**Issue by jk-olaoluwa** [wazuh/wazuh#17337]
|Wazuh version| Component | 
|---| --- |
| 4.10.0 | SCA | 

Rework SCA Policy for Microsoft Windows Server 2012 (R2)

- [ ] Solve typos.
- [ ] Increase check coverage. 
- [ ] Improve rules implementation.
- [ ] Improve texts.
- [ ] Solve reported issues.
       - [ ] Fix the check "Ensure 'Domain member: Maximum machine account password age' is set to '30 or fewer days, but not 0'. (Automated)"
      - [ ] Fix the check "Ensure 'Accounts: Guest account status' is set to 'Disabled' (MS only). (Automated)"
      - [ ] https://github.com/wazuh/external-devel-requests/issues/523

**Comment by Johnng007** [comment]
Fix for Issue https://github.com/wazuh/wazuh/issues/21298


  # (L1) Ensure 'Domain member: Maximum machine account password age' is set to '30 or fewer days, but not 0'. (Automated)
  - id: 15521
    title: "Ensure 'Domain member: Maximum machine account password age' is set to '30 or fewer days, but not 0'."
    description: "This policy setting determines the maximum allowable age for a computer account password. By default, domain members automatically change their domain passwords every 30 days. If you increase this interval significantly so that the computers no longer change their passwords, an attacker would have more time to undertake a brute force attack against one of the computer accounts. The recommended state for this setting is: 30 or fewer days, but not 0. Note: A value of 0 does not conform to the benchmark as it disables maximum password age. Note #2: Some problems can occur as a result of machine account password expiration, particularly if a machine is reverted to a previous point-in-time state, as is common with virtual machines. Depending on how far back the reversion is, the older machine account password stored on the machine may no longer be recognized by the domain controllers, and therefore the computer loses its domain trust. This can also disrupt non-persistent VDI implementations, and devices with write filters that disallow permanent changes to the OS volume. Some organizations may choose to exempt themselves from this recommendation and disable machine account password expiration for these situations."
    rationale: "In Active Directory-based domains, each computer has an account and password just like every user. By default, the domain members automatically change their domain password every 30 days. If you increase this interval significantly, or set it to 0 so that the computers no longer change their passwords, an attacker will have more time to undertake a brute force attack to guess the password of one or more computer accounts."
    impact: "None - this is the default behavior."
    remediation: "To establish the recommended configuration via GP, set the following UI path to 30 or fewer days, but not 0: Computer Configuration\\Policies\\Windows Settings\\Security Settings\\Local Policies\\Security Options\\Domain member: Maximum machine account password age."
    references:
      - "https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/domain-member-maximum-machine-account-password-age"
    compliance:
      - cis: ["2.3.6.5"]
      - cis_csc_v8: ["5.2"]
      - cmmc_v2.0: ["IA.L2-3.5.7"]
      - pci_dss_v4.0: ["2.2.2", "8.3.5", "8.3.6", "8.6.3"]
      - soc_2: ["CC6.1"]
    condition: all
    rules:
	
	WRONG RULE
      - 'c:net.exe accounts -> n:Maximum password age (days):\s+(\d+) compare <= 30'
      - 'c:net.exe accounts -> n:Maximum password age (days):\s+(\d+) compare > 0'
	  
	CORRECT RULE
      - 'r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Netlogon\Parameters -> MaximumPasswordAge'
      - 'r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Netlogon\Parameters -> MaximumPasswordAge -> 30'
      - 'not r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Netlogon\Parameters -> MaximumPasswordAge -> 0'
	  
	  
	  
	  
  # (L1) Ensure 'Accounts: Guest account status' is set to 'Disabled'. (Automated)
  - id: 15509
    title: "Ensure 'Accounts: Guest account status' is set to 'Disabled'."
    description: "This policy setting determines whether the Guest account is enabled or disabled. The Guest account allows unauthenticated network users to gain access to the system. The recommended state for this setting is: Disabled. Note: This setting will have no impact when applied to the Domain Controllers organizational unit via group policy because Domain Controllers have no local account database. It can be configured at the domain level via group policy, similar to account lockout and password policy settings."
    rationale: "The default Guest account allows unauthenticated network users to log on as Guest with no password. These unauthorized users could access any resources that are accessible to the Guest account over the network. This capability means that any network shares with permissions that allow access to the Guest account, the Guests group, or the Everyone group will be accessible over the network, which could lead to the exposure or corruption of data."
    impact: "All network users will need to authenticate before they can access shared resources. If you disable the Guest account and the Network Access: Sharing and Security Model option is set to Guest Only, network logons, such as those performed by the Microsoft Network Server (SMB Service), will fail. This policy setting should have little impact on most organizations because it is the default setting in Microsoft Windows 2000, Windows XP, and Windows Server™ 2003."
    remediation: "To establish the recommended configuration via GP, set the following UI path to Disabled: Computer Configuration\\Policies\\Windows Settings\\Security Settings\\Local Policies\\Security Options\\Accounts: Guest account status."
    references:
      - "https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/accounts-guest-account-status"
    compliance:
      - cis: ["2.3.1.2"]
      - cis_csc_v8: ["4.7"]
      - cis_csc_v7: ["16.8"]
      - iso_27001-2013: ["A.9.2.1"]
      - pci_dss_v3.2.1: ["2.1", "2.1.1"]
      - pci_dss_v4.0: ["2.2.2", "2.3.1"]
      - soc_2: ["CC6.3"]
    condition: any
    rules:
      - 'c:net user guest -> r:Account active\s+No'
      - "c:net user guest -> r:The user name could not be found."

**Comment by Johnng007** [comment]
**05-02-2024**

- Preparing SCA test environment
- Generating yaml file from the CIS pdf file

**Comment by jk-olaoluwa** [comment]
### Update 02/04/2024
 - Setting up testing environment

**Comment by Johnng007** [comment]
**Update 04/04/2024**
All fixed https://github.com/wazuh/wazuh/commit/41ec99a5c7a5d516e1dfe504e58166e19a17247c

**Comment by ooniagbi** [comment]
Currently working on other issues.

**Comment by ooniagbi** [comment]
This is still on hold because of higher-priority issues.

**Comment by ooniagbi** [comment]
This is still on hold because of higher-priority issues.

**Comment by IsExec** [comment]
# Windows Server 2012 R2 Rework SCA Testing Report
**Condition**: The new SCA was benchmarked against old Windows server 2012 R2 SCA.  

## Agent running on old SCA (Windows server 2012 R2 SCA)

###  Endpoint spec

<details>
<summary>CPU/RAM</summary>

````console
>systeminfo

Host Name:                 WINDOWS-SERVER-
OS Name:                   Microsoft Windows Server 2012 R2 Standard
OS Version:                6.3.9600 N/A Build 9600
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Server
OS Build Type:             Multiprocessor Free
Registered Owner:          
Registered Organization:   Vagrant
Product ID:                00252-00105-69793-AA339
Original Install Date:     1/13/2015, 3:27:02 AM
System Boot Time:          7/31/2024, 12:53:04 PM
System Manufacturer:       innotek GmbH
System Model:              VirtualBox
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 142 Stepping 10 GenuineIntel ~2112 Mhz
BIOS Version:              innotek GmbH VirtualBox, 12/1/2006
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-06:00) Central Time (US & Canada)
Total Physical Memory:     4,024 MB
Available Physical Memory: 2,936 MB
Virtual Memory: Max Size:  5,432 MB
Virtual Memory: Available: 4,489 MB
Virtual Memory: In Use:    943 MB
Page File Location(s):     C:\pagefile.sys
````

</details>

<details>
<summary>Disk</summary>

````console
>wmic logicaldisk get size,freespace,caption
Caption  FreeSpace    Size         
C:       39383060480  64055406592  
D:  
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/60534ca3-6b41-46a4-8045-c3a4a3849c92)
![disk_percent_plot](https://github.com/user-attachments/assets/12ed9415-5959-478d-824d-1c72d36c6df1)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/6efc4a83-37fa-4ee7-bcc1-48aee2751e16)
![memory_percent_plot](https://github.com/user-attachments/assets/73f46562-3786-4ba6-b612-8180744896a8)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/b8fe2b4b-99eb-4431-b3be-9db047ba7622)
![ProcessesCPU](https://github.com/user-attachments/assets/d16f2014-959a-4ad1-910f-f4e29b078324)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/0da60903-5ace-4d0f-b687-8a7e4dbe2704)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/e0689760-ce0b-4908-9b08-ab1b8cdee16f)
![ProcessesRAM](https://github.com/user-attachments/assets/e6172686-b03c-48b1-a0c3-2f9b57cd3e52)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/f4fcdbd1-4fe4-4262-bc99-d75e9c95b8e2)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/a3d422b8-3164-454e-96ff-664b46c4dfd7)
![swap_percent_plot](https://github.com/user-attachments/assets/795d4d1d-a9bf-4d10-a02d-3c1c761e82c7)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/b9d5bd29-8ce9-41cf-966a-1adeaa9347b2)
</details>

<details>
<summary>SCA logs</summary>

````console
2024/07/31 15:13:28 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/31 15:13:28 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/07/31 15:13:31 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/07/31 15:13:31 sca: INFO: Security Configuration Assessment scan finished. Duration: 3 seconds.
2024/07/31 15:18:28 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/31 15:18:28 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/07/31 15:18:31 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/07/31 15:18:31 sca: INFO: Security Configuration Assessment scan finished. Duration: 3 seconds.
2024/07/31 15:23:28 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/31 15:23:28 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
````

</details>


[metrics.zip](https://github.com/user-attachments/files/16450177/metrics.zip)



## Agent running on new SCA (Windows server 2019 SCA Rework)

### Endpoint spec
<details>
<summary>CPU/RAM</summary>

````console
>systeminfo

Host Name:                 WINDOWS-SERVER-
OS Name:                   Microsoft Windows Server 2012 R2 Standard
OS Version:                6.3.9600 N/A Build 9600
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Server
OS Build Type:             Multiprocessor Free
Registered Owner:          
Registered Organization:   Vagrant
Product ID:                00252-00105-69793-AA339
Original Install Date:     1/13/2015, 3:27:02 AM
System Boot Time:          7/31/2024, 4:38:06 PM
System Manufacturer:       innotek GmbH
System Model:              VirtualBox
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 142 Stepping 10 GenuineIntel ~2112 Mhz
BIOS Version:              innotek GmbH VirtualBox, 12/1/2006
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-06:00) Central Time (US & Canada)
Total Physical Memory:     4,024 MB
Available Physical Memory: 2,731 MB
Virtual Memory: Max Size:  5,432 MB
Virtual Memory: Available: 4,293 MB
Virtual Memory: In Use:    1,139 MB
Page File Location(s):     C:\pagefile.sys
````

</details>

<details>
<summary>Disk</summary>

````console
>wmic logicaldisk get size,freespace,caption
Caption  FreeSpace    Size         
C:       40912048128  64055406592  
D:                                 
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/9acbfcac-6802-4769-8614-a8865fff2eb6)
![disk_percent_plot](https://github.com/user-attachments/assets/9de464a9-145f-48fc-96e8-486c165e3375)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/ab5076e2-2b21-454b-9473-3289428b7f18)
![memory_percent_plot](https://github.com/user-attachments/assets/d4395da9-4893-42cd-874d-bc419bcd2ae4)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/17c6ba57-f3f7-4734-874c-f887597928f8)
![ProcessesCPU](https://github.com/user-attachments/assets/92faabb3-f88e-40c0-9d8b-e2f50c904b0e)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/419edfc8-8c52-4291-bd41-02fa5f3ab57a)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/00e9c2a6-ae4a-4357-a6e3-835a4bda32d4)
![ProcessesRAM](https://github.com/user-attachments/assets/3231893c-002a-44dc-a551-dccd5a15b304)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/53513472-a908-45c7-bfda-7817dcedd7d7)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/7f3427d3-f86f-4d2e-9cdd-6334d13b92fe)
![swap_percent_plot](https://github.com/user-attachments/assets/78cff161-9906-48dc-9b98-ce1b74fe1085)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/39076e2f-7b9a-48cf-bdc2-45b77e62bf3b)
</details>

<details>
<summary>SCA logs</summary>

````console
2024/07/31 22:20:50 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/31 22:20:51 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/07/31 22:20:56 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/07/31 22:20:56 sca: INFO: Security Configuration Assessment scan finished. Duration: 6 seconds.
2024/07/31 22:25:50 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/31 22:25:50 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/07/31 22:25:56 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/07/31 22:25:56 sca: INFO: Security Configuration Assessment scan finished. Duration: 6 seconds.
2024/07/31 22:30:50 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/31 22:30:50 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/07/31 22:30:56 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/07/31 22:30:56 sca: INFO: Security Configuration Assessment scan finished. Duration: 6 seconds.
````

</details>

[metrics.zip](https://github.com/user-attachments/files/16450230/metrics.zip)

## Analysis


  | Benchmark (Win   server 2012 R2 old SCA) | New SCA (Win   server 2012 R2 rework) | Difference
-- | -- | -- | --
CPU |   |   |  
`wazuh-agent.exe` average   CPU percent | 0.6 | 2.3 | 1.7
Overall   average CPU percent | 0.4 | 11.6 | 11.2
Overall peak   CPU percent | 0.5 | 18.1 | 17.6
Memory |   |   | 0
`wazuh-agent.exe`   average memory (MB) | 0.02 | 0.02 | 0
Overall   average memory used (MB) | 1084.86 | 1299.9 | 215.04
Overall peak   memory used peak (MB) | 1085.29 | 1301.05 | 215.76
Disk |   |   | 0
Overall   average disk used (MB) | 23530.62 | 22069.54 | -1461.08
Overall peak   disk used (MB) | 23530.62 | 22069.54 | -1461.08
Overall   average disk read bytes | N/A | N/A | N/A
Overall peak   disk read bytes | N/A | N/A | N/A
Overall   average disk write bytes | N/A | N/A | N/A
Overall peak   disk write bytes | N/A | N/A | N/A

## Note on Disk Read Bytes and Disk Write Bytes
Disk read and disk write bytes were excluded because the get_metrics.py script returned an Attribute error.

<details>
<summary>Error</summary>
```console
>python get_metrics.py -- interval 5
-- duration 600
Traceback <most recent call last):
File "C:\Users\Administrator\Desktop\sca_footprint\get_metrics.py", line 164,
in <module>
global_metrics = get_global_env_values (args.unit>

File "C:\Users\Administrator\Desktop\sca_footprint\get_metrics.py", line 95, i
n get_global_env_values
disk_read_bytes = convert_units<disk_io.read_bytes, unit>
AAA
AttributeError: 'NoneType' object has no attribute 'read_bytes'
```

</details>

THis was excluded by commenting the parts of the script that are supposed to fetch the disk read and write byes data.

**Comment by IsExec** [comment]
# Observations

## Invalid checks
The Old SCA resulted in 177 invalid checks out of 233 that were skipped during the SCA run due to missing registry keys.
This accounted for the shorter time of completion (3 seconds) and lower CPU utilization.
However, this will affect the accuracy of the SCA file as reported by users previously. 


<details>
<summary>Old SCA report</summary>
```console
DEBUG: Sending event: {"type":"summary","scan_id":18588516,"name":"CIS Microsoft Windows Server 2012 R2 Benchmark v2.3.0","policy_id":"cis_win2012r2","file":"cis_win2012r2.yml","description":"This document provides prescriptive guidance for establishing a secure configuration posture for Microsoft Windows Server 2012 R2.","references":"https://www.cisecurity.org/cis-benchmarks/","passed":36,"failed":21,"invalid":177,"total_checks":234,"score":63.157894134521484,"start_time":1722485532,"end_time":1722485535,
````
</details>

<details>
<summary>Rework SCA report</summary>
```console
DEBUG: Sending event: {"type":"summary","scan_id":2089258166,"name":"CIS Microsoft Windows Server 2012 R2 Benchmark v3.0.0","policy_id":"cis_win2012r2","file":"cis_win2012r2.yml","description":"This document provides prescriptive guidance for establishing a secure configuration posture for Microsoft Windows Server 2012 R2.","references":"https://www.cisecurity.org/cis-benchmarks/","passed":82,"failed":209,"invalid":0,"total_checks":291,"score":28.178693771362305,"start_time":1722484118,"end_time":1722484124,
````
</details>

**Comment by IsExec** [comment]
Windows Server 2012 R2 Rework SCA Re-Test Report


## Note on Disk Read Bytes and Disk Write Bytes
To capture the disk write and disk read values, it is required to enable disk metrics by running the command below:

```console
>diskperf -y
```
**Condition**: The new SCA was benchmarked against old Windows server 2012 R2 SCA.  

## Agent running on old SCA (Windows server 2012 R2 SCA)

###  Endpoint spec

<details>
<summary>CPU/RAM</summary>

````console
>systeminfo

Host Name:                 WINDOWS-SERVER-
OS Name:                   Microsoft Windows Server 2012 R2 Standard
OS Version:                6.3.9600 N/A Build 9600
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Server
OS Build Type:             Multiprocessor Free
Registered Owner:          
Registered Organization:   Vagrant
Product ID:                00252-00105-69793-AA339
Original Install Date:     1/13/2015, 3:27:02 AM
System Boot Time:          8/11/2024, 1:07:05 PM
System Manufacturer:       innotek GmbH
System Model:              VirtualBox
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 142 Stepping 10 GenuineIntel ~2112 Mhz
BIOS Version:              innotek GmbH VirtualBox, 12/1/2006
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-06:00) Central Time (US & Canada)
Total Physical Memory:     4,024 MB
Available Physical Memory: 3,126 MB
Virtual Memory: Max Size:  5,432 MB
Virtual Memory: Available: 4,560 MB
Virtual Memory: In Use:    872 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP

````

</details>

<details>
<summary>Disk</summary>

````console
>wmic logicaldisk get size,freespace,caption
C a p t i o n     F r e e S p a c e         S i z e                   
 
 C :               4 3 3 1 7 2 4 8 0 0 0     6 4 0 5 5 4 0 6 5 9 2   
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/5c490be2-62ab-467a-b516-b863d79e761e)
![disk_percent_plot](https://github.com/user-attachments/assets/d53fd2dd-0b20-4fb2-8c23-665ba2e49221)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/be4bba7e-af37-4cab-8590-0de9b454f99b)
![memory_percent_plot](https://github.com/user-attachments/assets/92e17466-236f-4487-a69b-4b103bc3e341)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/83e23eb2-8749-45d7-b005-2f96341c2833)
![ProcessesCPU](https://github.com/user-attachments/assets/981837a6-8b80-4e82-881a-71bfee6deb26)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/e643889f-735c-4958-aec1-c873efa04433)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/0314ed2e-8267-48c8-b350-4dc81667b7df)
![ProcessesRAM](https://github.com/user-attachments/assets/67b79262-ca57-4052-a81a-b65d3b2bb406)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/be02651f-6a5b-4c84-ba9c-72ccfb8fe645)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/40139181-4ffd-4919-834b-f2e0b08e76d5)
![swap_percent_plot](https://github.com/user-attachments/assets/9d86216f-254c-48ee-ab72-9b2cdd9b1b61)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/10a9b1db-b35c-477d-b589-68c29a6bb331)
</details>

<details>
<summary>SCA logs</summary>

````console
2024/08/11 12:40:27 sca: INFO: Starting Security Configuration Assessment scan.
2024/08/11 12:40:27 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/08/11 12:40:31 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/08/11 12:40:31 sca: INFO: Security Configuration Assessment scan finished. Duration: 4 seconds.
2024/08/11 12:45:27 sca: INFO: Starting Security Configuration Assessment scan.
2024/08/11 12:45:27 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/08/11 12:45:30 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/08/11 12:45:30 sca: INFO: Security Configuration Assessment scan finished. Duration: 3 seconds.
2024/08/11 12:50:27 sca: INFO: Starting Security Configuration Assessment scan.
2024/08/11 12:50:27 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/08/11 12:50:30 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/08/11 12:50:30 sca: INFO: Security Configuration Assessment scan finished. Duration: 3 seconds.
````

</details>

[metrics.zip](https://github.com/user-attachments/files/16576206/metrics.zip)


## Agent running on new SCA (Windows server 2012 R2 SCA Rework)

### Endpoint spec
<details>
<summary>CPU/RAM</summary>

````console
>systeminfo

Host Name:                 WINDOWS-SERVER-
OS Name:                   Microsoft Windows Server 2012 R2 Standard
OS Version:                6.3.9600 N/A Build 9600
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Server
OS Build Type:             Multiprocessor Free
Registered Owner:          
Registered Organization:   Vagrant
Product ID:                00252-00105-69793-AA339
Original Install Date:     1/13/2015, 3:27:02 AM
System Boot Time:          8/11/2024, 1:07:05 PM
System Manufacturer:       innotek GmbH
System Model:              VirtualBox
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 142 Stepping 10 GenuineIntel ~2112 Mhz
BIOS Version:              innotek GmbH VirtualBox, 12/1/2006
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-06:00) Central Time (US & Canada)
Total Physical Memory:     4,024 MB
Available Physical Memory: 3,126 MB
Virtual Memory: Max Size:  5,432 MB
Virtual Memory: Available: 4,560 MB
Virtual Memory: In Use:    872 MB
Page File Location(s):     C:\pagefile.sys

````

</details>

<details>
<summary>Disk</summary>

````console
>wmic logicaldisk get size,freespace,caption
C a p t i o n     F r e e S p a c e         S i z e                   
 
 C :               4 3 3 1 7 2 4 8 0 0 0     6 4 0 5 5 4 0 6 5 9 2     
 
 D :                                
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/4be98c91-7fcd-4c5e-9715-cd84f07342e8)
![disk_percent_plot](https://github.com/user-attachments/assets/76e99355-c5f4-49a8-a8df-8bda4e93e037)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/8137d0f3-bdde-46e7-a876-8a10277b99d7)
![memory_percent_plot](https://github.com/user-attachments/assets/30d8e30c-3404-4792-9ce2-1f449b64c6dd)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/ff312a21-254f-463f-988b-5a2fe20fbdf1)
![ProcessesCPU](https://github.com/user-attachments/assets/cdb0e6f6-852e-4202-8156-97dc6645bf1b)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/c0fdc7a1-57e1-443e-b0e8-ecf68c16e2c1)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/28de53d4-0958-48ca-b741-641bc7266703)
![ProcessesRAM](https://github.com/user-attachments/assets/32cfe7d5-3060-4640-9732-c333eb2d4d2f)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/d025bbeb-6ab8-4329-8405-5a3911f4e509)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/a8eec920-8126-49f4-bf53-05af188f5b37)
![swap_percent_plot](https://github.com/user-attachments/assets/d98032e2-492b-494e-adaf-8ce1a305f06b)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/b50b150c-988b-49c7-ab05-117f3cf6c398)
</details>

<details>
<summary>SCA logs</summary>

````console
2024/08/11 13:45:11 sca: INFO: Starting Security Configuration Assessment scan.
2024/08/11 13:45:11 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/08/11 13:45:15 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/08/11 13:45:15 sca: INFO: Security Configuration Assessment scan finished. Duration: 4 seconds.
2024/08/11 13:50:11 sca: INFO: Starting Security Configuration Assessment scan.
2024/08/11 13:50:11 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/08/11 13:50:15 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/08/11 13:50:15 sca: INFO: Security Configuration Assessment scan finished. Duration: 4 seconds.
2024/08/11 13:55:11 sca: INFO: Starting Security Configuration Assessment scan.
2024/08/11 13:55:12 sca: INFO: Starting evaluation of policy: 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/08/11 13:55:15 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\cis_win2012r2.yml'
2024/08/11 13:55:15 sca: INFO: Security Configuration Assessment scan finished. Duration: 4 seconds.
````

</details>

[metrics.zip](https://github.com/user-attachments/files/16576213/metrics.zip)


## Analysis


  | Benchmark (Win   server 2012 R2 SCA) | New SCA (Win   server 2012 nonR2) | Difference
-- | -- | -- | --
CPU |   |   |  
`wazuh-agent.exe` average   CPU percent | 1.2 | 0.75 | -0.45
Overall   average CPU percent | 0.4 | 3.1 | 2.7
Overall peak   CPU percent | 0.8 | 3.6 | 2.8
Memory |   |   | 0
`wazuh-agent.exe`   average memory (MB) | 0.02 | 0.02 | 0
Overall   average memory used (MB) | 1162.15 | 943.53 | -218.62
Overall peak   memory used peak (MB) | 1162.23 | 944.19 | -218.04
Disk |   |   | 0
Overall   average disk used (MB) | 19193.59 | 20285.03 | 1091.44
Overall peak   disk used (MB) | 19193.59 | 20285.04 | 1091.45
Overall   average disk read bytes | 1019.89 | 1.86 | -1018.03
Overall peak   disk read bytes | 1019.89 | 1.86 | -1018.03
Overall   average disk write bytes | 342.36 | 19.74 | -322.62
Overall peak   disk write bytes | 342.45 | 19.94 | -322.51

**Comment by ooniagbi** [comment]
LGTM!

**Comment by ooniagbi** [comment]
Merged to 4.10.0
